tests/CMakeLists.txt

```diff
@@ -11,6 +11,37 @@ target_include_directories(utils INTERFACE
     .
 )
 
+function(cuke_set_environment environment)
+    set(options)
+    set(oneValueArgs )
+    set(multiValueArgs RUNPATH BINPATH)
+    
+    cmake_parse_arguments(CUKE_ENV "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+    
+    if(NOT "${CUKE_ENV_UNPARSED_ARGUMENTS}" STREQUAL "")
+        message(
+            FATAL_ERROR
+                "unparsed arguments in call to cuke_set_environment: ${CUKE_ENV_UNPARSED_ARGUMENTS} from ${CMAKE_CURRENT_LIST_FILE}"
+        )
+    endif()
+
+    if( NOT "${CUKE_ENV_RUNPATH}" STREQUAL "")
+        if(WIN32)
+            string(REPLACE ";" "\;" runpath "${CUKE_ENV_RUNPATH}")
+        endif()
+        set(RUNPATH "$<IF:$<PLATFORM_ID:Windows>,PATH,LD_LIBRARY_PATH>")
+        list(APPEND environment "$<IF:$<PLATFORM_ID:Windows>,PATH,LD_LIBRARY_PATH>=path_list_prepend:$<SHELL_PATH:${runpath}>")
+    endif()
+    if( NOT "${CUKE_ENV_BINPATH}" STREQUAL "")
+        if(WIN32)
+            string(REPLACE ";" "\;" binpath "${CUKE_ENV_BINPATH}")
+        endif()
+        list(APPEND environment "PATH=path_list_prepend:$<SHELL_PATH:${binpath}>")
+    endif()
+    
+    set(${environment} ${${environment}} PARENT_SCOPE)
+endfunction()
+
```

Comment:
Why is this needed? is it to run the tests under windows with dynamic linked cucumber libraries?
It bothers me that so much cmake code is needed, I fear it is brittle and gets difficult to maintain. I'd like to see a more straight forward solution (i.e. more of a standard solution, less cmake code) if possible.

Reply:
running the tests on windows would not be possible if qt is not in PATH. I do not know or found any code how this is set in your workflow file.
This is what the cmake code is for.
I can reduce it to RUNPATH handling only. But if not there i have a strong feeling that the tests would not run on windows.